src/server/v2.0/handler/user.go

```diff
@@ -0,0 +1,485 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/goharbor/harbor/src/common"
+	"github.com/goharbor/harbor/src/common/rbac"
+	"github.com/goharbor/harbor/src/common/rbac/system"
+	"github.com/goharbor/harbor/src/common/security"
+	"github.com/goharbor/harbor/src/common/security/local"
+	"github.com/goharbor/harbor/src/common/utils"
+	"github.com/goharbor/harbor/src/controller/config"
+	"github.com/goharbor/harbor/src/controller/user"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/permission/types"
+	usermodels "github.com/goharbor/harbor/src/pkg/user/models"
+	"github.com/goharbor/harbor/src/server/v2.0/handler/model"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	"github.com/goharbor/harbor/src/server/v2.0/restapi"
+	"github.com/goharbor/harbor/src/server/v2.0/restapi/operations/users"
+)
+
+var userResource = system.NewNamespace().Resource(rbac.ResourceUser)
+
+type usersAPI struct {
+	BaseAPI
+	ctl     user.Controller
+	getAuth func(ctx context.Context) (string, error) // For testing
+}
+
+func (u *usersAPI) SetCliSecret(ctx context.Context, params users.SetCliSecretParams) middleware.Responder {
+	uid := int(params.UserID)
+	if err := u.requireForCLISecret(ctx, uid); err != nil {
+		return u.SendError(ctx, err)
+	}
+	if err := requireValidSecret(params.Secret.Secret); err != nil {
+		return u.SendError(ctx, err)
+	}
+	if err := u.ctl.SetCliSecret(ctx, uid, params.Secret.Secret); err != nil {
+		logger.Errorf("Failed to set CLI secret, error: %v", err)
```

Comment:
`log.G(ctx).Errorf("Failed to set CLI secret, error: %v", err)`?

Reply:
fixed